tap_mysql/connection.py

```diff
@@ -99,16 +99,28 @@ def __init__(self, config):
         # that the connection fails, the patch is reverted by reassigning the
         # patched out method to it's original spot.
 
-        args = {
-            "user": config["user"],
-            "password": config["password"],
-            "host": config["host"],
-            "port": int(config["port"]),
-            "cursorclass": config.get("cursorclass") or pymysql.cursors.SSCursor,
-            "connect_timeout": CONNECT_TIMEOUT_SECONDS,
-            "read_timeout": READ_TIMEOUT_SECONDS,
-            "charset": "utf8",
-        }
+        if config["unix_socket"]:
+            args = {
+                "user": config["user"],
+                "password": config["password"],
+                "unix_socket": config['unix_socket'],
+                "cursorclass": config.get("cursorclass") or pymysql.cursors.SSCursor,
+                "connect_timeout": CONNECT_TIMEOUT_SECONDS,
+                "read_timeout": READ_TIMEOUT_SECONDS,
+                "charset": "utf8",
+            }
+        else:
+            args = {
+                "user": config["user"],
+                "password": config["password"],
+                "host": config["host"],
+                "port": int(config["port"]),
+                "cursorclass": config.get("cursorclass") or pymysql.cursors.SSCursor,
+                "connect_timeout": CONNECT_TIMEOUT_SECONDS,
+                "read_timeout": READ_TIMEOUT_SECONDS,
+                "charset": "utf8",
+            }
+
```

Comment:
Move the shared config outside of the if/else so it's clear what the if/else is changing. (untested code below, my python is rusty)

```suggestion
        args = {
            "user": config["user"],
            "password": config["password"],
            "cursorclass": config.get("cursorclass") or pymysql.cursors.SSCursor,
            "connect_timeout": CONNECT_TIMEOUT_SECONDS,
            "read_timeout": READ_TIMEOUT_SECONDS,
            "charset": "utf8",
        }
        if config["unix_socket"]:
            args["unix_socket"] = config['unix_socket']
        else:
            args.update({
                "host": config["host"],
                "port": int(config["port"]),
            })
```